### ISSUE TITLE
MiniProfiler crashes intermittently.

### ISSUE THREAD

**Issue by sarojkh** [MiniProfiler/rack-mini-profiler#354]
Lately MiniProfiler has been crashing every now and then, and I can't seem to reproduce the behavior at will. I am not familiar with the intricacies of this gem.

This has been noticed after upgrading to `v1.0.0` (previously when the application had `v0.10.5`, this issue was never observed).

```
Unexpected error while processing request: can't convert nil into Integer
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rack-mini-profiler-1.0.0/lib/mini_profiler/timer_struct/page.rb:73:in `%'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rack-mini-profiler-1.0.0/lib/mini_profiler/timer_struct/page.rb:73:in `extra_json'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rack-mini-profiler-1.0.0/lib/mini_profiler/timer_struct/page.rb:64:in `to_json'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rack-mini-profiler-1.0.0/lib/mini_profiler/profiler.rb:104:in `serve_results'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rack-mini-profiler-1.0.0/lib/mini_profiler/profiler.rb:129:in `serve_html'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rack-mini-profiler-1.0.0/lib/mini_profiler/profiler.rb:178:in `call'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/railties-5.0.6/lib/rails/engine.rb:522:in `call'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/railties-5.0.6/lib/rails/railtie.rb:193:in `public_send'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/railties-5.0.6/lib/rails/railtie.rb:193:in `method_missing'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/connection.rb:86:in `block in pre_process'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/connection.rb:84:in `catch'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/connection.rb:84:in `pre_process'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/connection.rb:53:in `process'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/connection.rb:39:in `receive_data'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/eventmachine-1.2.5/lib/eventmachine.rb:194:in `run_machine'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/eventmachine-1.2.5/lib/eventmachine.rb:194:in `run'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/backends/base.rb:73:in `start'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/server.rb:162:in `start'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/controllers/controller.rb:87:in `start'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/runner.rb:203:in `run_command'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/lib/thin/runner.rb:159:in `run!'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thin-1.7.2/bin/thin:6:in `<top (required)>'
	/Users/whatever/.rbenv/versions/2.5.1/bin/thin:23:in `load'
	/Users/whatever/.rbenv/versions/2.5.1/bin/thin:23:in `<top (required)>'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `load'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `kernel_load'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:28:in `run'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:424:in `exec'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:27:in `dispatch'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:18:in `start'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/exe/bundle:30:in `block in <top (required)>'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/friendly_errors.rb:122:in `with_friendly_errors'
	/Users/whatever/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/exe/bundle:22:in `<top (required)>'
	/Users/whatever/.rbenv/versions/2.5.1/bin/bundle:23:in `load'
	/Users/whatever/.rbenv/versions/2.5.1/bin/bundle:23:in `<main>'


```

From what I can tell, this is caused by a change in https://github.com/MiniProfiler/rack-mini-profiler/pull/333 by @superiorlu.

https://github.com/MiniProfiler/rack-mini-profiler/blob/ab2d911d2c17aa15c2fba0be8edc0fbfea00997a/lib/mini_profiler/timer_struct/page.rb#L71-L76

Before the aforementioned PR, the line 73 was
```
 :started               => '/Date(%d)/' % @attributes[:started],
```

This has something to do with the absence of key-value pair for key `:started_at`; probably because it was deleted in an earlier call (This is just a hunch by looking at the code. I seriously don't know intricacies of this gem).

**Comment by gregmolnar** [comment]
This one can be closed, it is fixed by #360

**Comment by SamSaffron** [comment]
thanks @gregmolnar !
